cmd/admissioncontroller/handlers.go

```diff
@@ -0,0 +1,379 @@
+		req.SubResource != "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		log.Printf("Field mismatch")
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "batch"
+	podschema.Version = "v1"
+	podschema.Kind = "Job"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*batch.Job)
+
+	pod := new(core.Pod)
+	ds.Spec.Template.Spec.DeepCopyInto(&pod.Spec)
+	ds.Spec.Template.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	ac.checkPodSpec(pod, ds.Namespace, w, req.UID)
+}
+
+func (ac *admissionController) handleCronJob(w http.ResponseWriter, r *http.Request) {
+	req, errcode := getAdmissionReviewRequest(r)
+	if errcode != 0 {
+		http.Error(w, http.StatusText(http.StatusUnsupportedMediaType), http.StatusUnsupportedMediaType)
+		return
+	}
+	if req.UID == "" ||
+		req.Kind.Group != "batch" || req.Kind.Version != "v1beta1" || req.Kind.Kind != "CronJob" ||
+		req.Resource.Group != "batch" || req.Resource.Version != "v1beta1" || req.Resource.Resource != "cronjobs" ||
+		req.SubResource != "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		log.Printf("Field mismatch")
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "batch"
+	podschema.Version = "v1beta1"
+	podschema.Kind = "CronJob"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*batch.CronJob)
+
+	pod := new(core.Pod)
+	ds.Spec.JobTemplate.Spec.Template.Spec.DeepCopyInto(&pod.Spec)
+	ds.Spec.JobTemplate.Spec.Template.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	ac.checkPodSpec(pod, ds.Namespace, w, req.UID)
+}
+
+func (ac *admissionController) handleDeploymentConfig(w http.ResponseWriter, r *http.Request) {
+	req, errcode := getAdmissionReviewRequest(r)
+	if errcode != 0 {
+		http.Error(w, http.StatusText(http.StatusUnsupportedMediaType), http.StatusUnsupportedMediaType)
+		return
+	}
+	if req.UID == "" ||
+		req.Kind.Group != "apps" || req.Kind.Version != "v1" || req.Kind.Kind != "DeploymentConfig" ||
+		req.Resource.Group != "apps" || req.Resource.Version != "v1" || req.Resource.Resource != "deploymentconfigs" ||
+		req.SubResource != "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		log.Printf("Field mismatch")
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "apps"
+	podschema.Version = "v1"
+	podschema.Kind = "DeploymentConfig"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*appsv1.DeploymentConfig)
+
+	pod := new(corev1.Pod)
```

Comment:
do all this once inside checkPodSpec.  Pass a corev1.PodSpec into checkPodSpec and construct a Pod from it in there.